src/enforcement/enforcer.ts

```diff
@@ -123,13 +124,17 @@ export interface IEnforcer {
 export class Enforcer implements IEnforcer {
   public contextStore: ContextStore; // cross-query context (global context)
   private client: AxiosInstance;
+  private opaClient: AxiosInstance;
 
   /**
    * Creates an instance of the Enforcer class.
    * @param config - The configuration object for the Permit SDK.
    * @param logger - The logger instance for logging.
    */
   constructor(private config: IPermitConfig, private logger: Logger) {
+    const opaBaseUrl = new URL(this.config.pdp);
```

Comment:
Maybe extract this logic to a function then write some Unit Tests for it, it can get flaky and complicated sometimes